python/src/numerous/_client/_fs_client.py

```diff
@@ -313,23 +314,28 @@ def get_collection_documents(
 
     def get_collection_file(
         self, collection_id: str, file_key: str
-    ) -> Optional[CollectionFileReference]:
+    ) -> Optional[NumerousFile]:
+        file_key = self._modify_id(file_key)
         path = self._base_path / collection_id / f"{file_key}.json"
         if not path.exists():
             return None
 
         file = FileSystemCollectionFile.load(path)
 
         file_id = str(Path(collection_id) / file_key)
-        return CollectionFileReference(
-            id=file_id,
+        self.file_to_path_map[file_id] = path
+        return NumerousFile(
+            client=self,
+            file_id=file_id,
             key=file_key,
-            downloadURL=os.fspath(file.path),
-            uploadURL=os.fspath(file.path),
-            tags=[tag.to_file_reference_tag() for tag in file.tags],
+            exists=True,
+            numerous_file_tags=[tag.to_file_reference_tag() for tag in file.tags],
         )
 
-    def delete_collection_file(self, file_id: str) -> Optional[CollectionFileReference]:
+    def _modify_id(self, file_key: str) -> str:
+        return f"file_{file_key}"
```

Comment:
This method does not seem to have anything to do with IDs. Maybe:
```suggestion
    def _fs_file_name(self, file_key: str) -> str:
        return f"file_{file_key}.json"
```